src/app/calculator/utilities/natural-gas-reduction/natural-gas-reduction.service.ts

```diff
@@ -0,0 +1,425 @@
+        tmpWaterMassFlowData = {
+          waterFlow: this.convertUnitsService.value(tmp.waterMassFlowData.waterFlow).from('L/s').to('gpm'),
+          inletTemperature: this.convertUnitsService.value(tmp.waterMassFlowData.inletTemperature).from('C').to('F'),
+          outletTemperature: this.convertUnitsService.value(tmp.waterMassFlowData.outletTemperature).from('C').to('F'),
+          systemEfficiency: tmpWaterMassFlowData.systemEfficiency
+        };
+        tmpOtherMethodData = {
+          consumption: this.convertUnitsService.value(tmp.otherMethodData.consumption).from(settings.energyResultUnit).to('MMBtu')
+        };
+      }
+
+      tmpData[i] = {
+        operatingHours: tmp.operatingHours,
+        fuelCost: tmp.fuelCost,
+        measurementMethod: tmp.measurementMethod,
+        flowMeterMethodData: tmpFlowMeterMethodData,
+        otherMethodData: tmpOtherMethodData,
+        airMassFlowData: tmpAirMassFlowData,
+        waterMassFlowData: tmpWaterMassFlowData,
+        units: tmp.units
+      };
+    }
+    let inputObj: NaturalGasReductionInput = {
+      naturalGasReductionInputVec: tmpData
+    };
+    let results: NaturalGasReductionResults = calculatorAddon.naturalGasReduction(inputObj);
+    if (settings.unitsOfMeasure == 'Metric') {
+      results = {
+        energyUse: this.convertUnitsService.value(results.energyUse).from('MMBtu').to(settings.energyResultUnit),
+        energyCost: results.energyCost,
+        annualEnergySavings: results.annualEnergySavings,
+        costSavings: results.costSavings,
+        heatFlow: this.convertUnitsService.value(results.heatFlow).from('MMBtu').to(settings.energyResultUnit),
+        totalFlow: settings.unitsOfMeasure == 'Metric' ? this.convertUnitsService.value(results.totalFlow).from('ft3/h').to('m3/h') : results.totalFlow
+      };
+    }
+    return results;
+  }
+
+  calculateIndividualEquipment(input: NaturalGasReductionData, settings: Settings): NaturalGasReductionResults {
+    let inputArray: Array<NaturalGasReductionData> = [input];
+    //need loop to support conversion
+    for (let i = 0; i < inputArray.length; i++) {
+      let tmpFlowMeterMethodData: FlowMeterMethodData = inputArray[i].flowMeterMethodData;
+      let tmpAirMassFlowMeasuredData: AirMassFlowMeasuredData = inputArray[i].airMassFlowData.airMassFlowMeasuredData;
+      let tmpAirMassFlowNameplateData: AirMassFlowNameplateData = inputArray[i].airMassFlowData.airMassFlowNameplateData;
+      let tmpAirMassFlowData: AirMassFlowData = inputArray[i].airMassFlowData;
+      let tmpWaterMassFlowData: WaterMassFlowData = inputArray[i].waterMassFlowData;
+      let tmpOtherMethodData: OtherMethodData = inputArray[i].otherMethodData;
+      let tmp = inputArray[i];
+      if (settings.unitsOfMeasure == 'Metric') {
+        tmpFlowMeterMethodData = {
+          flowRate: this.convertUnitsService.value(tmp.flowMeterMethodData.flowRate).from('m3/h').to('ft3/h')
+        };
+        tmpAirMassFlowMeasuredData = {
```

Comment:
Pulling the logic for the conversion to it's own function will allow you to use it here as well. Create functions instead of copy and pasting...